hadoop-crypto/src/test/java/com/palantir/crypto2/hadoop/EncryptedFileSystemTest.java

```diff
@@ -266,4 +266,67 @@ public void testGetCipherAlgorithm_nonDefault() throws IOException, URISyntaxExc
         assertThat(efs.getCipherAlgorithm(), is(cipherAlg));
     }
 
+    @Test
+    public void testDelete_successful() throws IOException {
+        testDeleteFile(false);
+    }
+
+    @Test
+    public void testDelete_recusiveDeleteOnFile() throws IOException {
+        testDeleteFile(true);
+    }
+
+    @Test
+    public void testDelete_recursiveDeleteOnDir() throws IOException {
+        OutputStream os = efs.create(path);
+        os.write(0x00);
+        os.close();
+
+        Path folderPath = new Path(folder.getRoot().getAbsolutePath());
+        assertTrue(efs.delete(folderPath, true));
+
+        assertFalse(efs.exists(path));
```

Comment:
Can't - the key material would only be deleted as a side effect of the recursive delete if you're using the file based strategy. Recursive deletes won't remove in-memory key material unless we want to add some logic to look at directory prefixes in the key storage map. The test uses the in-memory strategy.